### PR TITLE
refactor: add trimetMode prop to toggle arrows and colors in transiti…

### PR DESCRIPTION
**Pull request by lightsage88**
This accomplishes adding a prop called 'trimetMode', a boolean to the transitive-overlay so that a layer can be programatically toggled on or off to display direcitonal arrows shower riders the direction they ought to be travelling towards on their given itinerary plan whether they are on foot or in transit. Additionally, several color changes are included that rely on the presence of the 'trimetMode' boolean to come into play for this feature. Nothing that was already there is otherwise tinkered with.

The latest commit includes a couple of type adjustments that allow for the package to build locally. I got a bunch of errors when I tried running
```
yarn prepublish
```
so I could double check my work.

### CODE REVIEW

File: packages/types/src/index.ts [hunk 1]
```diff
@@ -97,25 +97,25 @@ export type ZoomBasedSymbol = {
...
-  tripId?: string;
+  stopId?: string;
   vehicleId?: string;
+  tripId?: string;
+  blockId?: string;
 
-  heading?: number;
   lat?: number;
   lon?: number;
+  heading?: number;
```
Comment (miles-grant-ibigroup) [inline packages/types/src/index.ts:118]: Why were these reordered? Was anything added/removed?

File: packages/types/src/index.ts [hunk 2]
```diff
@@ -545,6 +545,7 @@ export type TransitiveJourney = {
     streetEdges: number[];
     to: TransitivePlace;
     type: string;
+    route_color?: string;
```
Comment (miles-grant-ibigroup) [inline packages/types/src/index.ts:548]: Why was this added?

File: packages/types/src/index.ts [hunk 3]
```diff
@@ -223,8 +223,8 @@ export type Step = {
   area: boolean;
   bogusName: boolean;
   distance: number;
-  elevation?: ElevationData;
-  elevationProfile?: ElevationProfileComponent[];
+  elevation: ElevationData;
+  elevationProfile: any;
```
[the remark this answers is not in the document]
Comment (lightsage88) [reply packages/types/src/index.ts:227]: Ah, this was done because I before I pushed i did a git pull origin master to keep things up to date. I found that I was getting build errors for things I didn't touch. So the second commit in this pull request is all stuff that I did _after_ I did everything I needed to do to get the feature working.

* bf90e3efdbef3857f1a59df4f7571fcaf4b5203a
* 5b1f90d703e4ae4b93e8c95eced81f5d644c65fd

When I finished my work, I ran:
`
git pull origin master
`
to make sure that I'm incorporating the latest and greatest into my branch so I'm not behind. Before this, running 
`
yarn prepublish
`
yielded no problems for me.

After getting my branch up to date and trying the same, however, I ran into this error:
`
lerna ERR! yarn run tsc exited 2 in '@opentripplanner/core-utils'
lerna ERR! yarn run tsc stdout:
$ tsc
src/itinerary.ts(6,3): error TS2305: Module '"@opentripplanner/types"' has no exported member 'ElevationProfileComponent'.
src/itinerary.ts(348,12): error TS2339: Property 'elevationProfile' does not exist on type 'Step'.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
`
I hadn't originally touched any of these files.

File: packages/transitive-overlay/src/index.tsx [hunk 4]
```diff
@@ -88,14 +89,28 @@ const accessLegFilter = [
...
+    trimetModeProps || {};
   const { current: map } = useMap();
 
+  const calculateColor = segmentData => {
+    let color;
+    if (ignoreRouteColor) {
+      color = segmentData?.route_color || trimetTeal;
+    } else {
+      color = modeColorMap[segmentData.type] || "#008";
+    }
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/index.tsx:110]: I don't think this quite works the way we expect it to! What do you think about something like this?
```suggestion
      let color = modeColorMap[segmentData.type] || defaultColorOverride || "#008";
      if (!ignoreRouteColor) {
        color = segmentData?.route_color;
    }
```

Comment (miles-grant-ibigroup) [reply packages/transitive-overlay/src/index.tsx:110]: Also I'm a little bit confused about the `route_color` feature here in general... These are street walking legs! Why are we giving them route colors.

File: packages/transitive-overlay/src/index.tsx [hunk 5]
```diff
@@ -88,14 +89,28 @@ const accessLegFilter = [
...
+  trimetModeProps: TrimetModeProps;
 };
 
 const TransitiveCanvasOverlay = ({
   activeLeg,
-  transitiveData
+  transitiveData,
+  trimetModeProps
 }: Props): JSX.Element => {
+  const { showRouteArrows = null, ignoreRouteColor = null, trimetTeal = null } =
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/index.tsx:100]: instead of making `trimetTeal` a boolean, we could make this more generic and call this `defaultColorOverride` instead

File: packages/transitive-overlay/src/index.tsx [hunk 6]
```diff
@@ -88,14 +89,28 @@ const accessLegFilter = [
 type Props = {
   activeLeg?: Leg;
   transitiveData?: TransitiveData;
+  trimetModeProps: TrimetModeProps;
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/transitive-overlay/src/index.tsx:92]: Also please make them optional

File: packages/transitive-overlay/src/index.tsx (hunk 4 above)
Comment (miles-grant-ibigroup) [reply packages/transitive-overlay/src/index.tsx:110]: Also, if we do simplify this function, I would make it inline again

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
We're getting there! Just a few things I'm still confused about

File: packages/transitive-overlay/src/index.tsx [hunk 7]
```diff
@@ -139,7 +145,9 @@ const TransitiveCanvasOverlay = ({
                       type: "Feature",
                       properties: {
                         type: "street-edge",
-                        color: modeColorMap[segment.type] || "#008",
+                        color: ignoreRouteColor
+                          ? segment?.route_color || defaultColorOverride
+                          : modeColorMap[segment.type] || "#008",
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/index.tsx:150]: Is this conditional flipped?

```suggestion
                        color: ignoreRouteColor
                          ? modeColorMap[segment.type] || "#008"
                          : segment?.route_color || defaultColorOverride,
```

Comment (miles-grant-ibigroup) [reply packages/transitive-overlay/src/index.tsx:150]: I'm also a little confused as to when `defaultColorOverride` is used.

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
I'm a little confused about the color override things that are going on, but otherwise we're getting there!

File: packages/transitive-overlay/src/index.tsx [hunk 8]
```diff
@@ -139,7 +147,9 @@ const TransitiveCanvasOverlay = ({
                       type: "Feature",
                       properties: {
                         type: "street-edge",
-                        color: modeColorMap[segment.type] || "#008",
+                        color: ignoreRouteColor
+                          ? defaultColorOverride
+                          : modeColorMap[segment.type] || "#008",
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/index.tsx:152]: Now that `defaultColorOverride` only does this, we can get rid of `ignoreRouteColor`!

File: packages/transitive-overlay/src/index.tsx [hunk 9]
```diff
@@ -208,7 +218,6 @@ const TransitiveCanvasOverlay = ({
   if (!transitiveData) return <></>;
 
   const { fromAnchor, toAnchor } = getFromToAnchors(transitiveData);
-
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/index.tsx:211]: Please undo this

File: packages/types/src/index.ts (hunk 2 above)
Comment (miles-grant-ibigroup) [inline packages/types/src/index.ts:548]: This can now be removed, right?

File: packages/transitive-overlay/src/index.tsx [hunk 10]
```diff
@@ -269,7 +278,54 @@ const TransitiveCanvasOverlay = ({
...
+            "symbol-placement": "line",
+            "icon-image": "arrow-icon",
+            "icon-size": 0.1,
+            "symbol-spacing": 10,
+            "icon-allow-overlap": true,
+            "icon-ignore-placement": true,
+            "icon-offset": [0, 8000]
+          }}
+          paint={{
+            "icon-color": ["get", "color"],
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/index.tsx:324]: Doesn't this work out to always be `routeColorOverride`? Why not just use that here and then you don't need `routeColorOverrideArrow`?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Getting close! Just some last cleanup

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Looking really good! Just have one comment left on line 324 of index.tsx

**Comment by miles-grant-ibigroup** [comment]
Doesn't this work out to always be routeColorOverride? Why not just use that here and then you don't need routeColorOverrideArrow?

File: packages/transitive-overlay/src/index.tsx (hunk 10 above)
Comment (lightsage88) [reply packages/transitive-overlay/src/index.tsx:324]: We want the routeColorOverrideArrow color to be there if there is no color to pull from: a 'default' in case there is no color to match up.

Comment (miles-grant-ibigroup) [reply packages/transitive-overlay/src/index.tsx:324]: Hmm... Should we use the `["get", "color"]` instead then? Is there a use case for the walk arrows being a different color than the other arrows?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]

File: packages/transitive-overlay/src/index.tsx (hunk 10 above)
Comment (lightsage88) [reply packages/transitive-overlay/src/index.tsx:324]: There is no use case and we've just agreed to drop the teal thing until it comes up later. Expect a new push soon! :)

Comment (lightsage88) [reply packages/transitive-overlay/src/index.tsx:324]: If it even comes up later.

File: packages/transitive-overlay/src/index.tsx [hunk 11]
```diff
@@ -139,7 +143,10 @@ const TransitiveCanvasOverlay = ({
                       type: "Feature",
                       properties: {
                         type: "street-edge",
-                        color: modeColorMap[segment.type] || "#008",
+                        color:
+                          defaultColorOverride ||
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/index.tsx:147]: Sorry last nitpick -- can we rename this just `colorOverride`? Since it now overrides every possible color

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
I think we're there! Also, could you please merge master into this branch? That should clear out the `yarn.lock` changes

**Review by miles-grant-ibigroup — APPROVED** [review]
We did it! Thanks so much for all your work. Before merging, please merge `master` into this branch

File: packages/transitive-overlay/src/index.tsx [hunk 12]
```diff
@@ -88,11 +88,15 @@ const accessLegFilter = [
 type Props = {
   activeLeg?: Leg;
   transitiveData?: TransitiveData;
+  showRouteArrows?: boolean;
```
Comment (binh-dam-ibigroup) [inline packages/transitive-overlay/src/index.tsx:91]: Sort props

File: packages/transitive-overlay/src/index.tsx [hunk 13]
```diff
@@ -88,11 +88,15 @@ const accessLegFilter = [
...
+  colorOverride?: string;
 };
 
 const TransitiveCanvasOverlay = ({
   activeLeg,
-  transitiveData
+  transitiveData,
+  showRouteArrows,
```
Comment (binh-dam-ibigroup) [inline packages/transitive-overlay/src/index.tsx:98]: Sort props

File: packages/transitive-overlay/src/index.tsx [hunk 14]
```diff
@@ -269,7 +274,25 @@ const TransitiveCanvasOverlay = ({
         }}
         type="line"
       />
-
+      {showRouteArrows && (
+        <Layer
+          id="route-arrows"
```
Comment (binh-dam-ibigroup) [inline packages/transitive-overlay/src/index.tsx:279]: Sort props

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Could you add stories please, one for each prop you introduced, to illustrate what the result should look like?

File: packages/transitive-overlay/src/index.tsx [hunk 15]
```diff
@@ -87,14 +88,48 @@ const accessLegFilter = [
...
   activeLeg?: Leg;
+  colorOverride?: string;
+  showRouteArrows?: boolean;
   transitiveData?: TransitiveData;
 };
 
+const images = [
+  {
+    url: routeArrow,
+    id: "arrow-icon"
```
Comment (binh-dam-ibigroup) [inline packages/transitive-overlay/src/index.tsx:99]: Sort attributes.

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks like a nice addition to transitive rendering. Thanks for adding the stories.

File: packages/transitive-overlay/src/index.tsx (hunk 15 above)
Comment (lightsage88) [reply packages/transitive-overlay/src/index.tsx:99]: I can get this swap, sure. But I'm having a hard time getting this through the CI/CD because of the stories. I had to hoist the arrow image into the branch along with the logic to add it to the map instance on runtime. The image is causing problems with the tests because it can't find the .png file needed for it even though when it's actually running it's fine.

Do you have any experience on resolving things like this? Miles sent me something about the storybook/main.js file to poke around in, but I'd appreciate all the hand holding I can get.

Comment (binh-dam-ibigroup) [reply packages/transitive-overlay/src/index.tsx:99]: Have you tried to update the snapshots? (CI runs node 16, so try with that version if you haven't).

Comment (lightsage88) [reply packages/transitive-overlay/src/index.tsx:99]: I believe I tried that. That's 
```
yarn test -u

OR

yarn unit -u
```
right?

Comment (binh-dam-ibigroup) [reply packages/transitive-overlay/src/index.tsx:99]: `yarn unit -u`

File: packages/transitive-overlay/src/index.tsx [hunk 16]
```diff
@@ -16,6 +16,8 @@ import { getRouteLayerLayout, patternToRouteFeature } from "./route-layers";
 import { drawArc, getFromToAnchors, itineraryToTransitive } from "./util";
 
 export { itineraryToTransitive };
+// import routeArrow from "./images/route_arrow.png";
+
```
Comment (binh-dam-ibigroup) [inline packages/transitive-overlay/src/index.tsx:20]: Remove this commented code.

Comment (lightsage88) [reply packages/transitive-overlay/src/index.tsx:20]: I'm still experimenting trying to figure out how to satisfy the CI. I'm bouncing back versions that have the arrows and don't to try and pinpoint what whether this is actually a webpack issue or a jest configuration issue. There's going to be some oddities for a while, but I'll get them all smoothed out for when it's all straightened out and flying right.

Comment (binh-dam-ibigroup) [reply packages/transitive-overlay/src/index.tsx:20]: From running `yarn unit -u` there are snapshot changes that need to go in. Can you try checking in the updated snapshots?

File: jest.config.js [hunk 17]
```diff
@@ -25,11 +25,12 @@ module.exports = {
...
-    "\\.(svg)$": "<rootDir>/__mocks__/file-mock.js"
+    "\\.(svg|png|jpg|jpeg)$": "<rootDir>/__mocks__/file-mock.js"
   },
   testPathIgnorePatterns: ["<rootDir>/packages/(?:.+?)/((esm)|(lib))/"],
   transform: {
     "\\.[jt]sx?$": "babel-jest",
     "\\.ya?ml$": "yaml-jest"
-  }
+  },
+  setupFilesAfterEnv: ["<rootDir>/jest.setup.js"]
```
Comment (miles-grant-ibigroup) [inline jest.config.js:35]: Why is this added?

File: jest.setup.js [hunk 18]
```diff
@@ -0,0 +1,6 @@
+// Mock window.URL.createObjectURL for Jest
+if (typeof window.URL.createObjectURL === "undefined") {
+  Object.defineProperty(window.URL, "createObjectURL", {
+    value: jest.fn()
+  });
+}
```
Comment (miles-grant-ibigroup) [inline jest.setup.js:6]: Do we need this file?

File: packages/itinerary-body/src/defaults/transit-leg-summary.tsx [hunk 19]
```diff
@@ -32,7 +32,7 @@ export default function TransitLegSummary({
       <S.InvisibleAdditionalDetails>
         <FormattedMessage
           defaultMessage={defaultMessages["otpUi.TransitLegBody.expandDetails"]}
-          description="Screen reader text to expand stops"
+          description="Screen reader text added to expand steps"
```
Comment (miles-grant-ibigroup) [inline packages/itinerary-body/src/defaults/transit-leg-summary.tsx:35]: Why was this changed?

File: packages/transitive-overlay/src/TransitiveOverlay.story.tsx [hunk 20]
```diff
@@ -337,3 +337,26 @@ export const OTP2ScooterItinerary = injectIntl(({ intl }) => (
   </>
 ));
 OTP2ScooterItinerary.decorators = [withMap([33.749, -84.388], 11)];
+
+export const ShowRouteArrows = (): JSX.Element => (
+  <>
+    <TransitiveOverlay
+      showRouteArrows
+      transitiveData={itineraryToTransitive(bikeOnlyItinerary, { companies })}
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/TransitiveOverlay.story.tsx:345]: Can we use an itinerary with transit instead?

File: packages/transitive-overlay/src/TransitiveOverlay.story.tsx [hunk 21]
```diff
@@ -337,3 +337,26 @@ export const OTP2ScooterItinerary = injectIntl(({ intl }) => (
...
+      visible
+    />
+  </>
+);
+ShowRouteArrows.decorators = [withMap([45.520441, -122.68302], 16)];
+
+export const ColorOverride = (): JSX.Element => (
```
Comment (miles-grant-ibigroup) [inline packages/transitive-overlay/src/TransitiveOverlay.story.tsx:352]: Should we maybe consider making these storybook controls instead of new stories?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
A few new cleanup things. Once those are done I'll update the snapshots and then we should be good to go!

File: jest.setup.js (hunk 18 above)
Comment (lightsage88) [reply jest.setup.js:6]: ...which leads us to this file, which seems to quell the error I would receive in the logs from attempts to get this whole thing through CI.

File: packages/transitive-overlay/src/TransitiveOverlay.story.tsx (hunk 21 above)
Comment (lightsage88) [reply packages/transitive-overlay/src/TransitiveOverlay.story.tsx:352]: I don't know what that is.

File: packages/transitive-overlay/src/TransitiveOverlay.story.tsx (hunk 20 above)
Comment (lightsage88) [reply packages/transitive-overlay/src/TransitiveOverlay.story.tsx:345]: Sure.

File: packages/itinerary-body/src/defaults/transit-leg-summary.tsx (hunk 19 above)
Comment (lightsage88) [reply packages/itinerary-body/src/defaults/transit-leg-summary.tsx:35]: I recall receiving some errors about this...let me dig into this further, but basically it was centered around there being different values for elements that were supposed to have the same thing. I updated to whichever entry was the most recent.

File: packages/transitive-overlay/src/TransitiveOverlay.story.tsx [hunk 22]
```diff
@@ -49,291 +50,144 @@ function getToLocation(itinerary) {
...
+  args: {
+    showRouteArrows: false,
+    colorOverride: null,
+    visible: true
+  }
 };
 
-export const Empty = (): JSX.Element => (
-  <TransitiveOverlay transitiveData={null} visible />
+const Template = injectIntl(
```
Comment (binh-dam-ibigroup) [inline packages/transitive-overlay/src/TransitiveOverlay.story.tsx:71]: Nice refactor!

File: packages/transitive-overlay/src/index.tsx [hunk 23]
```diff
@@ -87,14 +89,49 @@ const accessLegFilter = [
...
+  {
+    id: "arrow-icon",
+    url: routeArrow
+  }
+];
+
 const TransitiveCanvasOverlay = ({
   activeLeg,
+  colorOverride,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (binh-dam-ibigroup) [inline packages/transitive-overlay/src/index.tsx:107]: Remove this eslint exception.

File: packages/transitive-overlay/src/index.tsx [hunk 24]
```diff
@@ -87,14 +89,49 @@ const accessLegFilter = [
 
 type Props = {
   activeLeg?: Leg;
+  colorOverride?: string;
```
Comment (binh-dam-ibigroup) [inline packages/transitive-overlay/src/index.tsx:92]: Unlike `showRouteArrows`, `colorOverride` only applies to non-transit legs only. Is that intended? If so, could you rename `colorOverride` to `accessLegColorOverride`?

**Review by binh-dam-ibigroup — COMMENTED** [review]
Is `colorOverride` intended for access (i.e. non-transit) legs only?

File: packages/transitive-overlay/src/index.tsx (hunk 24 above)
Comment (lightsage88) [reply packages/transitive-overlay/src/index.tsx:92]: Sure, I can do that!

**Comment by binh-dam-ibigroup** [comment]
@lightsage88 Have you updated the snapshots? (`yarn unit -u`)

**Comment by lightsage88** [comment]
I can't do that from my machine. I thought we established that either you or Miles would clone this down and run that for me.

**Review by binh-dam-ibigroup — APPROVED** [review]
This looks good now, thanks for all the changes!

**Review by miles-grant-ibigroup — APPROVED** [review]
I'm a little worried about how many snapshots are changing, but everything seems to be still working
